docs/software/Containerizationon.md

```diff
@@ -15,14 +15,210 @@ Containers distinguish themselves through their low computational overhead and t
+
+Here we provide an example of using a gcc compiler with OpenMPI. Alpine uses an Infiniband interconnect. In order to use a Singularity container with OpenMPI (or any MPI) on Alpine, OpenMPI needs to be installed both inside and outside of the container. More specifically, the _same_ version of OpenMPI needs to be installed inside and outside (at least very similar, you can sometimes get away with two different minor versions, e.g. 2.1 and 2.0). 
+
+CURC can provide users with a recipe that ensures the appropriate version of OpenMPI is installed in the image. This recipe can be used as a template to build your own MPI-enabled container images for Alpine.
+
+Once you’ve built the container with one of the methods outlined above, you can place it on Alpine and run it on a compute node. The following is an example of running a gcc/OpenMPI container with Apptainer on Alpine. The syntax is a normal MPI run where multiple instances of a Singularity image are run. The following example runs `mpi_hello_world` with MPI from a container.
+
+```
+ml gcc/11.2.0
+ml openmpi/4.1.1
+
+mpirun -np 4 apptainer exec openmpi.sif mpi_hello_world"
+```
+### Useful Apptainer Features
+
+When using/constructing containers using Apptainer, there are a number of tools that users can deploy to ensure desired functionality. Features of high-importance are as follows: 
+
+* **Fakeroot**: Fakeroot is a feature which allows a non-root user to obtain nearly-root-level administrative permissions **only within the requisite container**. This affords users the ability to perform high-level operations within a container, such as: 
+
+<ol>
+  <li>Package installation using `sudo`</li>
+  <li>Changing user/group identity</li>
+  <li>Permissions changes</li>
+</ol> 
+
+Note that any changes made within the container with Fakeroot are not reflected outside of the container. Users can deploy fakeroot by running any of `shell`, `exec`, `run`, or `build` with the `--fakeroot` flag. For example:
+```
+apptainer build --fakeroot test.sif test.def
+```
+
+* **Fix-Perms**: Performing a container operation with the fix-perms flag will ensure that the user has read, write, and execute permissions for all container content from Open Container Initiative and Docker sources. Users can deploy fix-perms by running `build` with the `--fix-perms` flag. For example:
+```
+apptainer build --fix-perms test.sif test.def
+```
+
+* **Sandbox**: Building containers can be a time-consuming process of trial-and-error. For this reason, you may want to perform operations within the container and have those operations reflected in your container image. For this purpose, you can use the sandbox option. Users can deploy a sandbox by running `shell` with the `--sandbox` flag. For example: 
```

Comment:
```suggestion
* **Sandbox**: Building containers can be a time-consuming process if building from the definition file. This is because you may have build errors that need to be resolved, which requires that you rebuild the container multiple times. For this reason, you may want to perform iterative installs within the container first to ensure compatibility, then transfer these commands to a definition file. For this purpose, you can use the sandbox option. Users can construct a sandbox by using the `--sandbox` flag. For example: 
```